### PR TITLE
feat: added support for Fastify requests

### PR DESCRIPTION
**Pull request by AlexStansfield**
I ran into issues using this package as I'm using fastify which doesn't have the same request methods as express.

So I've created this PR. However I'm a little unsure if I went the right way with checking the request type. I looked for the `originalUrl` property which is only on Express.

I'm not happy with that solution but as there is no way to check something is an instance of an interface I'm not sure what else I can do. Any Suggestions for a better solution would be great.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
     "eslint": "^7.32.0",
     "eslint-config-prettier": "^8.3.0",
     "eslint-plugin-prettier": "^4.0.0",
+    "fastify": "^3.22.1",
```
[the remark this answers is not in the document]
Comment (ppetzold) [reply package.json:43]: @AlexStansfield ping, in case you missed it ;)

[the remark this answers is not in the document]
Comment (AlexStansfield) [reply package.json:43]: @ppetzold sorry, just saw this. The `FastifyRequest` type is used in the decorator test. If I remove Fastify from dev dependencies then the test won't run.

[the remark this answers is not in the document]
Comment (ppetzold) [reply package.json:43]: ah sorry, you are right :) thought for some reason it's a dep intead of devDep.

[the remark this answers is not in the document]
Comment (ppetzold) [reply package.json:43]: npm v7 brings back auto-install for peerDeps. may check this out later

**Review by ppetzold — APPROVED** [review]

**Pull request merged**